web/modules/custom/asset_distribution/asset_distribution.behat.inc

```diff
@@ -0,0 +1,354 @@
+    $result = $query->execute();
+
+    if (empty($result)) {
+      throw new \InvalidArgumentException("The asset distribution with the name '$name' was not found.");
+    }
+
+    return Rdf::load(reset($result));
+  }
+
+  /**
+   * Returns the Solution with the given name.
+   *
+   * If multiple solution have the same name, the first one will be returned.
+   *
+   * @param string $name
+   *   The solution name.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   The solution.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when a solution with the given name does not exist.
+   */
+  public function getSolutionByName($name) {
+    $query = \Drupal::entityQuery('rdf_entity')
+      ->condition('?entity', 'rdf:type', SparqlArg::uri('http://www.w3.org/ns/adms#Asset'))
+      ->condition('?entity', SparqlArg::uri('http://purl.org/dc/terms/title'), SparqlArg::literal($name))
+      ->range(0, 1);
+    $result = $query->execute();
+
+    if (empty($result)) {
+      throw new \InvalidArgumentException("The solution with the name '$name' was not found.");
+    }
+
+    // We need to skip the static cache and load the unchanged entity from the
+    // database.
+    return \Drupal::entityTypeManager()
+      ->getStorage('rdf_entity')
+      ->loadUnchanged(reset($result));
+  }
+
+  /**
+   * Checks the number of available asset distributions.
+   *
+   * @param int $number
+   *   The expected number of asset distributions.
+   *
+   * @throws \Exception
+   *   Thrown when the number of asset distributions does not
+   *   match the expectation.
+   *
+   * @Then I should have :number asset distribution(s)
+   */
+  public function assertAssetDistributionCount($number) {
+    $actual = \Drupal::entityQuery('rdf_entity')
+      ->condition('?entity', 'rdf:type', '<http://www.w3.org/ns/adms#AssetDistribution>')
+      ->count()
+      ->execute();
+    if ($actual != $number) {
+      throw new \Exception("Wrong number of asset distributions. Expected number: $number, actual number: $actual.");
+    }
+  }
+
+  /**
+   * Checks if the given asset distributions belongs to the given solution.
+   *
+   * If there are multiple solutions or asset distributions with
+   * the same name, then only the first one is checked.
+   *
+   * @param string $asset_distribution
+   *   The name of the asset distribution to check.
+   * @param string $solution
+   *   The title of the custom page to check.
+   *
+   * @throws \Exception
+   *   Thrown when an asset distribution with the given title doesn't exist.
+   *
+   * @Then the :asset_distribution asset distribution is related to the :solution solution
+   */
+  public function assertAssetRelationship($asset_distribution, $solution) {
+    /** @var \Drupal\rdf_entity\RdfInterface $asset_distribution */
+    $asset_distribution = $this->getAssetDistributionByName($asset_distribution);
+
+    /** @var \Drupal\rdf_entity\RdfInterface $solution */
+    $this->clearStaticCaches();
```

Comment:
That was actually leftover. Removed it as it is not needed anymore. I replaced the load function with the loadUnchanged so that we can get the latest version on loading of an entity.